From 3aefb5ee843fbe4789d03bb181e190d462df95e4 Mon Sep 17 00:00:00 2001
From: Luis Chamberlain <mcgrof@kernel.org>
Date: Wed, 3 Nov 2021 09:58:43 -0700
Subject: [PATCH] nvdimm/btt: do not call del_gendisk() if not needed
Git-commit: 3aefb5ee843fbe4789d03bb181e190d462df95e4
Patch-mainline: v5.16-rc1
References: git-fixes

del_gendisk() should not called if the disk has not been added. Fix this.

(Coly Li: rebased for Linux 4.12 based SUSE kernel)

Fixes: 41cd8b70c37a ("libnvdimm, btt: add support for blk integrity")
Reviewed-by: Dan Williams <dan.j.williams@intel.com>
Reviewed-by: Christoph Hellwig <hch@lst.de>
Signed-off-by: Luis Chamberlain <mcgrof@kernel.org>
Link: https://lore.kernel.org/r/20211103165843.1402142-1-mcgrof@kernel.org
Signed-off-by: Jens Axboe <axboe@kernel.dk>
Signed-off-by: Coly Li <colyli@suse.de>

---
 drivers/nvdimm/btt.c |    1 -
 1 file changed, 1 deletion(-)

--- a/drivers/nvdimm/btt.c
+++ b/drivers/nvdimm/btt.c
@@ -1555,7 +1555,6 @@ static int btt_blk_init(struct btt *btt)
 		int rc = nd_integrity_init(btt->btt_disk, btt_meta_size(btt));
 
 		if (rc) {
-			del_gendisk(btt->btt_disk);
 			put_disk(btt->btt_disk);
 			blk_cleanup_queue(btt->btt_queue);
 			return rc;
